packages/commercetools/theme/middleware/checkout.js

```diff
@@ -0,0 +1,37 @@
+import { getMe } from '@vue-storefront/commercetools-api';
+
+const canEnterPersonalDetails = cart => !cart.customerId;
+
+const canEnterShipping = cart => cart.customerEmail || cart.customerId;
+
+const canEnterPayment = cart => cart.shippingInfo && cart.shippingAddress;
+
+const canEnterReview = cart => cart.billingAddress;
+
+export default async ({ app }) => {
+  const currentPath = app.context.route.fullPath.split('/checkout/')[1];
+  const { data: { me: { activeCart } } } = await getMe();
+
+  switch (currentPath) {
+    case 'personal-details':
+      if (!canEnterPersonalDetails(activeCart)) {
+        app.context.redirect('/checkout/shipping');
+      }
+      break;
+    case 'shipping':
+      if (!canEnterShipping(activeCart)) {
+        app.context.redirect('/');
+      }
+      break;
+    case 'payment':
+      if (!canEnterPayment(activeCart)) {
+        app.context.redirect('/');
+      }
+      break;
+    case 'order-review':
+      if (!canEnterReview(activeCart)) {
+        app.context.redirect('/');
+      }
```

Comment:
shouldn't be here case for `/checkout` and `/checkout/thank-you` :thinking:

Reply:
it depends, in the future, we  have to cover two scenarios:
- enter without product, so you are not in the checkout flow, then  redirect to the home
- enter when you are in the checkout flow, but you haven't finished previous steps, then move  to the last step

for now, easy case, improve later